### ISSUE TITLE
:bug: MCBatchNorm: running statistics discarded during fitting

### ISSUE THREAD

**Issue by crlna16** [torch-uncertainty/torch-uncertainty#218]
Hello, thank you for this amazing library!

I'm encountering an issue with the `MCBatchNorm` wrapper. I'm applying it to a model that was already trained using standard `BatchNorm2d` layers. At inference time, I load the model weights and use the wrapper to replace all `BatchNorm2d` layers with `MCBatchNorm2d`.

The problem arises during the fitting of the `MCBatchNorm` layers:
The mean and variance are recalculated for each estimator using batches from a post-processing dataset, up to `n_estimator` times. However, this process completely discards the original running statistics learned during training. In my case, this leads to significantly different normalization behavior, and as a result, the predictions are qualitatively and quantitatively different from the ones with the original model.

**Code reference**:
[https://github.com/ENSTA-U2IS-AI/torch-uncertainty/blob/e9f63301d94a986d0e7c252ba9517fb8355f0d68/torch_uncertainty/layers/mc_batch_norm.py#L58C1-L59C56](https://github.com/ENSTA-U2IS-AI/torch-uncertainty/blob/e9f63301d94a986d0e7c252ba9517fb8355f0d68/torch_uncertainty/layers/mc_batch_norm.py#L58C1-L59C56)

According to Teye et al. ([https://arxiv.org/pdf/1802.06455](https://arxiv.org/pdf/1802.06455)), during MC sampling, each estimator should update the batch norm statistics using a batch from the training set:

> “Sampling ˆωᵢ therefore involves sampling a batch B from the training set and updating the parameters in the BN units, just as if we were taking a training step with B.”

This suggests that instead of discarding the original statistics, the `MCBatchNorm` layers should retain the running statistics from training and perform updates using batches from the training data. This would result in a set of n_estimator mean/var pairs that are slightly different but still based on the original running statistics. The current implementation seems to replace them outright, which may not reflect the intended behavior described in the paper.

Looking forward for your thoughts on this issue!

**Comment by o-laurent** [comment]
@alafage, if you have some time, I'd love to get your perspective on this. All in all, I'm pretty confident that the implementation is different from what I understand from the paper. + It would mean that the momentum of batch normalization is critical to the method's performance, whereas it is not mentioned at any point in the original paper, afaik. I plan to simplify the implementation of the MC Batch norm layer (stop using momentum and track stats params), and to provide warnings in all related docstrings and in the documentation.
